File: map/Interpreter2.0/DataStructure/MyStack.cs
using System.Collections.Generic;
using System;
namespace DataStructure
{
    public class MyStack<T> : MyIntStack<T>
    {
        Stack<T> stack;

        public MyStack(){
            stack = new Stack<T>();
        }


        public void push(T val) {
            stack.Push(val);
        }

       
        public T pop() {
            return stack.Pop();
        }

        public bool empty() {
            return stack.Count == 0;
        }

       
        override public String ToString() {
            String printstr = "";
            printstr += "ExecStack:";
            List<T> array = new List<T>(stack);
            array.Reverse();
            foreach(T e in array)
                printstr += "\n" + e.ToString();
            return printstr;
        }
        
    }
}

File: map/InterpreterFX/src/DataStructure/MyStack.java
package DataStructure;

import java.util.Collections;
import java.util.Stack;
import java.util.ArrayList;

public class MyStack<T> implements MyIntStack<T> {

    Stack<T> stack;

    public MyStack(){
        stack = new Stack<T>();
    }

    @Override
    public void push(T val) {
        stack.push(val);
    }

    @Override
    public T pop() {
        return stack.pop();
    }

    @Override
    public boolean empty() {
        return stack.empty();
    }

    @Override
    public String toString() {
        String printstr = new String();
        printstr += "ExecStack:";
        ArrayList<T> array = new ArrayList<T>(stack);
        Collections.reverse(array);
        for(T e: array)
            printstr += "\n" +e.toString();
        return printstr;
    }
}
